packages/harpguru-core/src/components/new-harpstrata-state-callback/hooks/use-derive-activedegreeids/use-derive-activedegreeids.ts

```diff
@@ -0,0 +1,15 @@
+import type { ActiveDegreeIds, HarpStrata } from 'harpstrata'
+
+import { deriveActiveDegreeIds } from '../../utils'
+import { compareActiveIds } from '../../../../utils/compare-active-ids'
+
+export const useDeriveActiveDegreeIds = (
+  newHarpStrata: HarpStrata,
+  prevActiveDegreeIds: ActiveDegreeIds,
+  setActiveDegreeIds: (arg0: ActiveDegreeIds) => void
+): ActiveDegreeIds => {
+  const nextActiveDegreeIds = deriveActiveDegreeIds(newHarpStrata)
+  if (!compareActiveIds(nextActiveDegreeIds, prevActiveDegreeIds))
+    setActiveDegreeIds(nextActiveDegreeIds)
+  return nextActiveDegreeIds
+}
```

Comment:
None of these are hooks. They're still just utils. And they probably all belong locally to the component that uses them unless they are used by both of the global state deriver components.

Reply:
Done